### PR TITLE
Update lv_chart.c

### PR DESCRIPTION
**Pull request by Rabdf**
Added function 'lv_chart_remove_series ()' which removes a series from a chart. Memory allocated by lvgl will be freed.

### CODE REVIEW

File: src/lv_widgets/lv_chart.c [hunk 1]
```diff
@@ -207,6 +207,27 @@ lv_chart_series_t * lv_chart_add_series(lv_obj_t * chart, lv_color_t color)
...
+ * @param chart pointer to a chart object
+ * @param ser   pointer to a data series on 'chart'
+ * @return true: successfully deleted
+ */
+bool lv_chart_remove_series(lv_obj_t * chart, lv_chart_series_t * ser)
+{
+    LV_ASSERT_OBJ(chart, LV_OBJX_NAME);
+    LV_ASSERT_NULL(ser);
+    
+    if(chart == NULL || ser == NULL) return;
```
Comment (embeddedt) [inline src/lv_widgets/lv_chart.c]: Needs `return false`.

File: src/lv_widgets/lv_chart.c [hunk 2]
```diff
@@ -207,6 +207,27 @@ lv_chart_series_t * lv_chart_add_series(lv_obj_t * chart, lv_color_t color)
     return ser;
 }
 
+/**
+ * Deallocate and remove a data series from a chart
+ * @param chart pointer to a chart object
+ * @param ser   pointer to a data series on 'chart'
+ * @return true: successfully deleted
+ */
+bool lv_chart_remove_series(lv_obj_t * chart, lv_chart_series_t * ser)
```
Comment (embeddedt) [inline src/lv_widgets/lv_chart.c]: Please add the function declaration to `lv_chart.h` as well.

**Review by embeddedt — APPROVED** [review]

File: src/lv_widgets/lv_chart.c [hunk 3]
```diff
@@ -207,6 +207,34 @@ lv_chart_series_t * lv_chart_add_series(lv_obj_t * chart, lv_color_t color)
...
```
Comment (kisvegabor) [inline src/lv_widgets/lv_chart.c]: Why we need a return value here? How can the deletion fail?

**Review by kisvegabor — COMMENTED** [review]
Thank you for the update. Would you be interested in adding `lv_chart_hide_series()` too?

**Comment by kisvegabor** [comment]
Seemingly the commits were messed up a little. 
"Added a function to hide/unhide a single series of a chart" is here but in #1952 too. In #1952 the CI error is also fixed.

**Comment by Rabdf** [comment]
The boolean return is for future changes. This way the prototype does not have to be changed if anything could fail then. But we may change the type of the return value to void.

Sorry for the mess-up of the uploads. This is the first time working with GitHub Desktop and some suggestions are missleading to me.

Rainer

07.12.2020 10:05:06 Gabor Kiss-Vamosi <notifications@github.com>:

**Comment by kisvegabor** [comment]
> But we may change the type of the return value to void.

I'd prefer the void value because the other functions look like this as well. 

> Sorry for the mess-up of the uploads. This is the first time working with GitHub Desktop and some suggestions are missleading to me.

No problem. Could you merge the hide/unhide branch here? I think it's easier than removing the last commit from here.

**Comment by Rabdf** [comment]
I think I finally made it. At least I learned at lot about GitHub and lvgl project.

Rainer

07.12.2020 14:53:26 Gabor Kiss-Vamosi <notifications@github.com>:

**Comment by kisvegabor** [comment]
Looks good, thank you!

**Comment by kisvegabor** [comment]
As it's a new feature, I'll merge it to master when the CI is ready.

**Comment by kisvegabor** [comment]
Finally, I cherry-picked the changes and added to dev manually because there were some ugly conflicts after changing the branch.

See  https://github.com/lvgl/lvgl/commit/dc8ef2bb6e576f5cd82049c52af69d9cdd9395d2

**Comment by kisvegabor** [comment]
Meanwhile, we made some reorganization on the branches. 
The gist is that the new feature is in `dev` which is based on v7 so it can directly replace `master`.

**Pull request closed without merging**